tensormap-client/src/components/DragAndDropCanvas/Canvas.js

```diff
@@ -0,0 +1,198 @@
+  }, []);
+
+  const modelData = (reactFlowInstance===null?{}:reactFlowInstance.toObject())
+  const modelValidateHandler = ()=>{
+    let data = {
+      "code": {
+          "dataset": {
+          "file_id": formState.selectedFile,
+          "target_field": formState.targetField,
+          "training_split": formState.trainTestRatio
+        },
+      "dl_model": {
+          "model_name": formState.modalName,
+          "optimizer": formState.optimizer,
+          "metric": formState.metric,
+          "epochs": formState.epochCount
+      },
+        "problem_type_id": formState.problemType
+    },
+    "model" : generateModelJSON(reactFlowInstance.toObject())
+    }
+
+     // Send the model data to the backend for validation and update the Modal state accordingly
+     validateModel(data)
+     .then(modelValidatedSuccessfully => {
+       setFormState(prevState => ({...prevState,
+           modelValidatedSuccessfully: modelValidatedSuccessfully,
+         }));
+       this.modelOpen();
+     })
+     .catch(error => {
+       console.error(error);
+       setFormState({ modelValidatedSuccessfully: false });
+       this.modelOpen();
+     });
+  }
+
+  const nodeTypes = useMemo(() => ({ custominput: InputNode,customdense:DenseNode,customflatten:FlattenNode }), [])
+  const onDrop = useCallback(
+    (event) => {
+      event.preventDefault();
+
+      const reactFlowBounds = reactFlowWrapper.current.getBoundingClientRect();
+      const type = event.dataTransfer.getData('application/reactflow');
+
+      // check if the dropped element is valid
+      if (typeof type === 'undefined' || !type) {
+        return;
+      }
+
+      const position = reactFlowInstance.project({
+        x: event.clientX - reactFlowBounds.left,
+        y: event.clientY - reactFlowBounds.top,
+      });
+      let newNode
+      if (type==='custominput'){
+        newNode = {
+          id: getId(),
+          type,
+          position,
+          data: { label: `${type} node`,params:{"dim-x":0,"dim-y":0} },
+        };
+      }else if(type==='customdense'){
+        newNode = {
+          id: getId(),
+          type,
+          position,
+          data: { label: `${type} node`,params:{"units":0, "activation":'default'} },
+        };
+      }else if (type==='customflatten'){
+        newNode = {
+          id: getId(),
+          type,
+          position,
+          data: { label: `${type} node`,params:{"dim-x":0,"dim-y":0}},
+        }}
+        else{
+          newNode = {
+            id: getId(),
+            type,
+            position,
+            data: { label: `${type} node`},
+          }
+        }
+      setNodes((nds) => nds.concat(newNode));
+      
+    },
+    [reactFlowInstance,setNodes]
+    );
+    
+    return (
+      <div>
+        <Grid celled='internally'>
+            <Grid.Row>
+              <Grid.Column width={13}>
+                <div className="dndflow" style={{ width: '60vw', height: '62vh' }}>
```

Comment:
Why we are giving inline styling if we have className? we can style this element using that className, can't we? 
Your thoughts?